src/cmd-remote-build

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python3 -u
+
+import sys
+import argparse
+from os import environ
+from subprocess import check_call, check_output, CalledProcessError
+
+def run_cmd(command, output=None):
+    '''
+    Run the given command using check_call and verify its return code.
+    @param str command command to be executed
+    '''
+    if output:
+        try:
+            output = check_output(command.split())
+        except CalledProcessError as e:
+            print('An exception has occurred: {0}'.format(e))
+            sys.exit(1)
+        return output
+    else:
+        try:
+            check_call(command.split())
+        except CalledProcessError as e:
+            print('An exception has occurred: {0}'.format(e))
+            sys.exit(1)
+
+def build(arch, commit, file, repo):
+    '''
+    Build the image  using podman remote and push to the registry
+    @param arch
+    @param commit
+    @param file
+    @param repo
+    @param tag
+    '''
+    tag = ("%s-%s" % (arch, commit))
+    run_cmd("podman --remote build --tag %s:%s-%s -f %s" % (repo, arch, commit, file))
+    run_cmd("podman --remote push %s:%s-%s" % (repo, arch, commit))
+
+    # Podman remote doesn't allow push using digestfile. That's why the tag check is done
+    tags = run_cmd("podman search --list-tags %s" % (repo), "output")
+    if (tag in str(tags)):
+        print("Build and Push done successfully via tag:%s" % (tag))
+    else:
+        print("Registry doesn't seem to have tag:%s" % (tag))
+
+def main():
+    parse_args()
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="CoreOS Assembler Remote Build",
+        description="Build coreos-assembler remotely",
+        usage="""
+Run multi-arch builds using podman remote.
+In order to get cmd-remote-build working the CONTAINER_SSHKEY and CONTAINER_HOST environment variables
+must be defined
+
+Examples:
+    $ cmd-remote-build \
+        --arch aarch64 \
+        --file src/DockerFile \
+        --repo quay.io/coreos/coreos-assembler-staging
+        --commit 31ffd """)
+
+    parser.add_argument(
+        '--arch', required=True,
+        help='Build Architecture')
+    parser.add_argument(
+        '--commit', required=True,
+        help='Repository commit used in the build')
+    parser.add_argument(
+        '--file', required=True,
+        help='Path to DockerFile')
+    parser.add_argument(
+        '--repo', required=True,
+        help='Registry repository')
```

Comment:
how about we add a `--push-to-registry` option that would enable pushing to the remote registry and document that the user needs to have previous run `podman login` for it to work. We would default to this being disabled but in the pipeline we would pass this option.

Could be useful for local testing to not push to a registry.

Reply:
+1